src/UserProfileService/Utilities/UserContextStore.cs

```diff
@@ -29,60 +28,21 @@ public UserContextStore(
     {
         _logger = logger;
         _context = contextAccessor.HttpContext;
-        _readService = readService;
-        _userId = new Lazy<Task<string>>(GetUserIdInternal);
+        _userId = new Lazy<string>(GetUserIdInternal);
```

Comment:
We called the asynchronous method in the constructor - that's why we were using the `Lazy<Task<string>>`.
Now this is not really needed any more. The GetUserIdInternal should be fast enough to be used directly in the constructor. 
It does not matter much, but I would remove it. "less code is better code" or "keep it simple".